### DISCUSSION TITLE
How to use Java code to delete graph indexes

### DISCUSSION THREAD

**Discussion by LiJie20190102** [JanusGraph/janusgraph#4326] [Q&A] [answered]
I created a mix index and wanted to use Java code to delete it. I then referred to AA for code writing, but encountered an error when deleting it. Is there a problem with my usage?

https://docs.janusgraph.org/schema/index-management/index-removal/
![image](https://github.com/JanusGraph/janusgraph/assets/53458004/80532b38-f409-4d6d-ba73-36a46dc28370)
![image](https://github.com/JanusGraph/janusgraph/assets/53458004/fe6b2859-34b6-4d97-9bda-e27f56369f3b)

**Comment by li-boxuan** [comment]
Can you try if https://docs.janusgraph.org/operations/recovery/#janusgraph-instance-failure works?

**Reply by LiJie20190102** [reply]
I can use the following code to delete it. I would like to ask if writing this way is consistent with [](https://docs.janusgraph.org/schema/index-management/index-removal/) Maintain consistency? If there are any issues with my code execution like this, I will try to fix the "Transaction Failure" issue in the way you said

![image](https://github.com/JanusGraph/janusgraph/assets/53458004/b15e65a9-1e27-4361-9cd7-d911b2e83b80)

**Reply by li-boxuan** [reply]
C.C. @rngcntr who developed DISCARD index feature, iirc

**Comment by rngcntr** [comment, marked as answer]
Hi @LiJie20190102!

The problem with your usage is that you are re-using the `graphIndex` object, which you created in a previous transaction. Any reference to an index only remains valid during the lifespan of the enclosing transaction. That's why the example you are following retrieves the index a second time by calling `m.getGraphIndex("name")`.
